config/v1/types_network.go

```diff
@@ -39,6 +40,20 @@ type NetworkSpec struct {
 	// Currently supported values are:
 	// - OpenShiftSDN
 	NetworkType string `json:"networkType"`
+
+	// externalIP is the policy applied to the ExternalIP field in a Service.
+	// If nil, any value is allowed for an ExternalIP. If the empty/zero
+	// policy is supplied, then ExternalIP is not allowed to be set.
+	// +optional
+	ExternalIPs *ExternalIPPolicy `json:"externalIPs,omitempty"`
+
+	// ingressIPs enable the use of automatically-assigned ExternalIPs for
+	// services on non-cloud-provider (i.e. bare-metal) clusters. If this
+	// is set, then services of type LoadBalancer will be automatically
+	// assigned an ExternalIP from this CIDR.
+	// Currently, only one value is supported
+	// +optional
+	IngressIPs []string `json:"ingressIPs,omitempty"`
```

Comment:
um... is there *another* PR associated with this? Because what you're describing here is not something the admission controller does.

Also, could we rename this since it does not refer to `Ingress.networking.k8s.io`? (And maybe "`CIDRs`" rather than "`IPs`" too?)